### PR TITLE
sanityCheck: only calculate siblingHash if necessary, refactor and reduce code nesting

### LINKED ISSUE #56: sanityCheck: only necessarily calculate siblingHash if len(proof.SideNodes > 0 but also return early and unnest
If we look at the code in here
https://github.com/celestiaorg/smt/blob/a99c0f5249884312ab8e6864fe165371c0f461ba/proofs.go#L44-L54

## Notice
* We are calculating a cryptographic hash even if len(proof.SideNodes)  conditions could cause a return https://github.com/celestiaorg/smt/blob/a99c0f5249884312ab8e6864fe165371c0f461ba/proofs.go#L46-L48
* Those nested conditions could be further simplified from
### Before
https://github.com/celestiaorg/smt/blob/a99c0f5249884312ab8e6864fe165371c0f461ba/proofs.go#L45-L52
### After
```go
        // Check that the sibling data hashes to the first side node if not nil
        if proof.SiblingData == nil {
                return true
        }
        
        if len(proof.SideNodes) == 0 {
                return true
        }
        
        siblingHash := th.digest(proof.SiblingData)
        return bytes.Equal(proof.SideNodes[0], siblingHash)
```

### PR DESCRIPTION
**Pull request by odeke-em**
Only calculate siblingHash to avoid wasting cycles when unneeded.
Also while here, reverse conditional the code to reduce nesting and
to make reading much simpler.

Fixes #56

### CODE REVIEW

**Comment by odeke-em** [comment]
Kindly cc-ing @cuonglm @adlerjohn @liamsi

File: proofs.go
Comment (cuonglm) [inline proofs.go:49]: Nit: maybe combine with above condition, so:

```
if proof.SiblingData == nil || len(proof.SideNodes) == 0 {
	return true
}
```

Also maybe write a simple test for this case, so the coverage isn't affected.

**Review by cuonglm — APPROVED** [review]

File: proofs.go
Comment (odeke-em) [reply proofs.go]: I've combined the conditions, thanks! For the test, it'll take quite a bit of context and time consuming which I'll delegate to the maintainers of this library :-) I am working on completing our audit currently.

**Review by liamsi — APPROVED** [review]
So much simpler to read. Thanks! 🚀 

I'll hold off merging until a second pair of eyes (either @adlerjohn or @musalbas) confirmed that the tiny drop in coverage is OK. IMO, it's OK to merge this as is.

**Review by adlerjohn — APPROVED** [review]

**Comment by adlerjohn** [comment]
Thanks @odeke-em!

**Pull request merged**
